From 4aa16c6eb090797f18c2a3ef73c21476189713d5 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:13:45 -0400
Subject: [alsa] hda: Don't set indep_hp flag for old AD codecs

Message-id: <140972823573-33-patchset-jkysela@redhat.com>
Patchwork-id: 90733
O-Subject: [RHEL 7.1 PATCH 032/442] ALSA: hda - Don't set indep_hp flag for old AD codecs
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Some old AD codecs don't like the independent HP handling, either it
contains a single DAC (AD1981) or it mandates the mixer routing
(AD1986A).  This patch removes the indep_hp flag for such codecs.

Bugzilla: https://bugzilla.kernel.org/show_bug.cgi?id=68081
Cc: <stable@vger.kernel.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: cbd209f41ea5f39394de5c1fe2dd9aa54a9c5744)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_analog.c b/sound/pci/hda/patch_analog.c
index 819d4a9..7a426ed 100644
--- a/sound/pci/hda/patch_analog.c
+++ b/sound/pci/hda/patch_analog.c
@@ -185,7 +185,7 @@ static const struct hda_codec_ops ad198x_auto_patch_ops = {
 };
 
 
-static int ad198x_parse_auto_config(struct hda_codec *codec)
+static int ad198x_parse_auto_config(struct hda_codec *codec, bool indep_hp)
 {
  struct ad198x_spec *spec = codec->spec;
  struct auto_pin_cfg *cfg = &spec->gen.autocfg;
@@ -195,7 +195,7 @@ static int ad198x_parse_auto_config(struct hda_codec *codec)
  codec->no_trigger_sense = 1;
  codec->no_sticky_stream = 1;
 
- spec->gen.indep_hp = 1;
+ spec->gen.indep_hp = indep_hp;
  spec->gen.add_stereo_mix_input = 1;
 
  err = snd_hda_parse_pin_defcfg(codec, cfg, NULL, 0);
@@ -379,7 +379,7 @@ static int patch_ad1986a(struct hda_codec *codec)
       ad1986a_fixups);
  snd_hda_apply_fixup(codec, HDA_FIXUP_ACT_PRE_PROBE);
 
- err = ad198x_parse_auto_config(codec);
+ err = ad198x_parse_auto_config(codec, false);
  if (err < 0) {
   snd_hda_gen_free(codec);
   return err;
@@ -481,7 +481,7 @@ static int patch_ad1983(struct hda_codec *codec)
 
  spec->gen.beep_nid = 0x10;
  set_beep_amp(spec, 0x10, 0, HDA_OUTPUT);
- err = ad198x_parse_auto_config(codec);
+ err = ad198x_parse_auto_config(codec, false);
  if (err < 0)
   goto error;
  err = ad1983_add_spdif_mux_ctl(codec);
@@ -568,7 +568,7 @@ static int patch_ad1981(struct hda_codec *codec)
  snd_hda_pick_fixup(codec, NULL, ad1981_fixup_tbl, ad1981_fixups);
  snd_hda_apply_fixup(codec, HDA_FIXUP_ACT_PRE_PROBE);
 
- err = ad198x_parse_auto_config(codec);
+ err = ad198x_parse_auto_config(codec, false);
  if (err < 0)
   goto error;
  err = ad1983_add_spdif_mux_ctl(codec);
@@ -894,7 +894,7 @@ static int patch_ad1988(struct hda_codec *codec)
  snd_hda_pick_fixup(codec, ad1988_fixup_models, NULL, ad1988_fixups);
  snd_hda_apply_fixup(codec, HDA_FIXUP_ACT_PRE_PROBE);
 
- err = ad198x_parse_auto_config(codec);
+ err = ad198x_parse_auto_config(codec, true);
  if (err < 0)
   goto error;
  err = ad1988_add_spdif_mux_ctl(codec);
@@ -1071,7 +1071,7 @@ static int patch_ad1884(struct hda_codec *codec)
  snd_hda_pick_fixup(codec, NULL, ad1884_fixup_tbl, ad1884_fixups);
  snd_hda_apply_fixup(codec, HDA_FIXUP_ACT_PRE_PROBE);
 
- err = ad198x_parse_auto_config(codec);
+ err = ad198x_parse_auto_config(codec, true);
  if (err < 0)
   goto error;
  err = ad1983_add_spdif_mux_ctl(codec);
@@ -1113,7 +1113,7 @@ static int patch_ad1882(struct hda_codec *codec)
  spec->gen.mixer_merge_nid = 0x21;
  spec->gen.beep_nid = 0x10;
  set_beep_amp(spec, 0x10, 0, HDA_OUTPUT);
- err = ad198x_parse_auto_config(codec);
+ err = ad198x_parse_auto_config(codec, true);
  if (err < 0)
   goto error;
  err = ad1988_add_spdif_mux_ctl(codec);
-- 
1.7.1